framework/archiver/__init__.py

```diff
@@ -0,0 +1,70 @@
+import json
+
+ARCHIVER_FAILURE = 'FAILURE'
+ARCHIVER_SUCCESS = 'SUCCESS'
+ARCHIVER_PENDING = 'ARCHIVING'
+ARCHIVER_CHECKING = 'CHECKING'
+ARCHIVER_SENDING = 'SENDING'
+ARCHIVER_SENT = 'SENT'
+
+class StatResult(object):
+    """
+    Helper class to collect metadata about a single file
+    """
+    num_files = 1
+
+    def __init__(self, target_id, target_name, disk_usage=0, meta=None):
+        self.target_id = target_id
+        self.target_name = target_name
+        self.disk_usage = float(disk_usage)
+        self.meta = meta
+
+    def __str__(self):
+        return json.dumps(self._to_dict())
+
+    def _to_dict(self):
+        return {
+            'target_id': self.target_id,
+            'target_name': self.target_name,
+            'disk_usage': self.disk_usage,
+            'meta': self.meta,
+        }
+
+
+class AggregateStatResult(object):
+    """
+    Helper class to collect metadata about aribitrary depth file/addon/node file trees
+    """
+    def __init__(self, target_id, target_name, targets=[], meta=None):
```

Comment:
`targets` should default to `None` or an empty tuple as the default list is mutable.